Fix OCID prefix regular expression.

jpmckinney opened the pull request:
- The trailing hyphen is not part of the OCID prefix.
- The OCID prefix is case-sensitive (and A-Z are not possible characters).
- The OCID prefix must prefix at least one character.

duncandewhurst commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Review and merge 'fix OCID prefix regular expression' PR](https://trello.com/c/rimkeDqc/71-review-and-merge-fix-ocid-prefix-regular-expression-pr)

ghost commented:
> The OCID prefix must prefix at least one character.

In order to do this, would a + not be better?

````
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import re
>>> his = re.compile(r'^ocds-[a-z0-9]{6}.?')
>>> mine =  re.compile(r'ocds-[a-z0-9]{6}.+')
>>> 
>>> 
>>> his.match('ocds-abc123')
<_sre.SRE_Match object; span=(0, 11), match='ocds-abc123'>
>>> mine.match('ocds-abc123')
>>> his.match('ocds-abc123-')
<_sre.SRE_Match object; span=(0, 12), match='ocds-abc123-'>
>>> mine.match('ocds-abc123-')
<_sre.SRE_Match object; span=(0, 12), match='ocds-abc123-'>

````

Also can this have an entry in the changelog? Thanks.

ghost commented:
> The OCID prefix must prefix at least one character.

Also, is there the possibility with this that we might end up with a bad UI? If this is used to report bad **prefixes** to the user, then "ocds-abc123" is a perfectly valid prefix, but not a valid ID.

jpmckinney commented:
> In order to do this, would a + not be better?

I'm not sure why I had a `?`, because `.?` can be omitted from a regex without change in behavior. I meant to just have a `.` (to test that the prefix is followed by a character). We don't need to match to the end of the string (which is what adding a `+` would do here). But:

> If this is used to report bad prefixes to the user, then "ocds-abc123" is a perfectly valid prefix

Good point, removed the `.` (`.?`).

> Also can this have an entry in the changelog?

Added.
